### ISSUE TITLE
Django 1.11.1 error

### ISSUE THREAD

**Issue by rob-vandam** [dokterbob/django-newsletter#230]
Django-newsletter with Djano-1.11.1 gives following error:

```
Request Method:	POST
Request URL:	https://mysite.com/newsletter/nieuwsbrief/subscribe/
Django Version:	1.11.1
Exception Type:	TypeError
Exception Value:	context must be a dict rather than Context.
Exception Location:	/home/koop/venv/lib/python3.4/site-packages/django/template/context.py in make_context, line 287
Python Executable:	/usr/local/bin/uwsgi
Python Version:	3.4.2
Python Path:	
['.',
 '',
 '/usr/lib/python3.4',
 '/usr/lib/python3.4/plat-x86_64-linux-gnu',
 '/usr/lib/python3.4/lib-dynload',
 '/home/koop/venv/lib/python3.4/site-packages']
```

**Comment by dokterbob** [comment]
Seems a bug indeed. Note that Django 1.11 is not (yet) officially supported.


https://docs.djangoproject.com/en/1.11/topics/templates/#django.template.backends.base.Template.render
"If context is provided, it must be a dict."

**Comment by dkdndes** [comment]
True for Django 1.11.2, too.

**Comment by rob-vandam** [comment]
In models.py I also had to replace:

`escaped_context = Context(variable_dict)`

With:

`escaped_contect = variable_dict`

Now it seems to work properly

**Comment by dokterbob** [comment]
That looks about right. Could you guys help the other users out with a pull request?

Make sure you enable tests for Django 1.11 in [.travis.yml](https://github.com/dokterbob/django-newsletter/blob/master/.travis.yml) so we can see whether or not this breaks compatibility. If it does, we need to find a way to support at least 1.8 too.

**Comment by dkdndes** [comment]
I tried to fix the issues as much as I was able to. I changed the context in admin.py and models.py and identified one error not related to Context.  

Then I fixed some of the 2to3 differences, applied isort and autopep8 and finally used the "pull request" to see if I travis discovers something when it runs against the different django versions. 

Unfortunatly it did discover two new issues. Not I am not sure how to handle these. 

Please have a look at [Pull Request](https://github.com/dokterbob/django-newsletter/pull/232) and the related Travis output. You see the issues here.

**Comment by thecardcheat** [comment]
I created #233 to address this issue specifically since #232 seemed to have a lot of unrelated changes that might delay acceptance. If there is something in #233 that is not sufficient, let me know, because all tests are passing.

**Comment by dokterbob** [comment]
#233 is merged now. Please test it and give feedback so that we might close this issue. If no feedback comes in a few weeks we can close this ticket regardless.

**Comment by fauziame** [comment]
I tried Replacing the escaped context fields, and now i get another error which states
No handlers could be found for logger "newsletter.models"

Why is that?

**Comment by dokterbob** [comment]
@thecardcheat @fauziame Could someone please confirm that this issue is fixed in master?

It'd be great to close this issue.

**Comment by thecardcheat** [comment]
I don't encounter this issue on master and I've been running a branch in production with #233 since I made the PR.

**Comment by lggwettmann** [comment]
I just encountered this issue on django-newsletter 0.6 and django 1.11.10 - any help on this?

**Comment by dokterbob** [comment]
What URL where you posting to? What were you trying to do, what was supposed to happen and what happened instead?

**Comment by dokterbob** [comment]
AFAIK it's fixed in 0.7b2

Guess it's time for a release :) 

Lars Wettmann <notifications@github.com> schreef op 9 juli 2018 04:06:17 CEST:
>I just encountered this issue on django-newsletter 0.6 and django
>1.11.10 - any help on this?

-- 
Verstuurd vanaf mijn Android apparaat met K-9 Mail. Excuseer mijn beknoptheid.

**Comment by lggwettmann** [comment]
yeah please. it's a major bug and i'm sure im not the only one experiencing it when installing it for the first time.
